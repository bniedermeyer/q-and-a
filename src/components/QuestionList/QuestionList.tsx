import { FunctionalComponent, h } from '@stencil/core';
import { AskedQuestion } from '../../models/Question';
import { incrementQuestionCount } from '../../utils/data-fetching-utils';
import state from '../../store';

interface QuestionListProps {
  questions: AskedQuestion[];
}

/**
 * For a given list of questions, this functional component will construct an unordered list and display them, along with a button to increment the count of each question.
 */
export const QuestionList: FunctionalComponent<QuestionListProps> = ({ questions }) => {
  if (questions.length === 0) {
    return null;
  }
  const questionsList = questions.map(question => (
    <li class="qa-question" key={question.key} style={{ fontSize: '0.75rem', margin: '5px', color: state.secondaryColor }}>
      <p class="qa-question-text" style={{ fontWeight: 'bold', wordWrap: 'break-word', margin: '3px' }}>
        {question.question}
      </p>
      <button
        type="button"
        class="qa-inc-button"
        onClick={() => incrementQuestionCount(question.key)}
        aria-label="Also ask this question"
        style={{ marginRight: '5px', border: `3px solid ${state.primaryColor}`, fontWeight: 'bold', backgroundColor: 'transparent' }}
      >
        +1
      </button>
      <span>{question.count}</span>
    </li>
  ));

  return (
    <ul id="questions" style={{ listStyle: 'none', paddingLeft: '0', padding: '3px', marginTop: '0', overflowY: 'scroll' }}>
      {questionsList}
    </ul>
  );
};
