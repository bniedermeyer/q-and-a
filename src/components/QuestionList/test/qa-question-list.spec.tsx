import {QuestionList} from '../QuestionList';


describe('qa-question-list', () => {
  it('renders no questions when none are present', async () => {
    const result = QuestionList({questions: []}, [], undefined);
    expect(result).toMatchSnapshot();
  });

  it('renders all of the questions', () => {
    const questions = [{
      key:'key1',
      question: 'What time is it?',
      count: 20
    }];
    
    const result = QuestionList({questions}, [], undefined);
    expect(result).toMatchSnapshot();
  })
});
