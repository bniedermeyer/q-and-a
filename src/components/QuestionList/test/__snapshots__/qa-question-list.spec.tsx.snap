// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`qa-question-list renders all of the questions 1`] = `
Object {
  "$attrs$": Object {
    "id": "questions",
    "style": Object {
      "listStyle": "none",
      "marginTop": "0",
      "padding": "3px",
      "paddingLeft": "0",
    },
  },
  "$children$": Array [
    Object {
      "$attrs$": Object {
        "class": "qa-question",
        "key": "key1",
        "style": Object {
          "color": "#112378",
          "fontSize": "0.75rem",
          "margin": "5px",
        },
      },
      "$children$": Array [
        Object {
          "$attrs$": Object {
            "class": "qa-question-text",
            "style": Object {
              "fontWeight": "bold",
              "margin": "3px",
              "wordWrap": "break-word",
            },
          },
          "$children$": Array [
            Object {
              "$attrs$": null,
              "$children$": null,
              "$elm$": null,
              "$flags$": 0,
              "$key$": null,
              "$name$": null,
              "$tag$": null,
              "$text$": "What time is it?",
            },
          ],
          "$elm$": null,
          "$flags$": 0,
          "$key$": null,
          "$name$": null,
          "$tag$": "p",
          "$text$": null,
        },
        Object {
          "$attrs$": Object {
            "aria-label": "Also ask this question",
            "class": "qa-inc-button",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "transparent",
              "border": "3px solid #10915b",
              "fontWeight": "bold",
              "marginRight": "5px",
            },
            "type": "button",
          },
          "$children$": Array [
            Object {
              "$attrs$": null,
              "$children$": null,
              "$elm$": null,
              "$flags$": 0,
              "$key$": null,
              "$name$": null,
              "$tag$": null,
              "$text$": "+1",
            },
          ],
          "$elm$": null,
          "$flags$": 0,
          "$key$": null,
          "$name$": null,
          "$tag$": "button",
          "$text$": null,
        },
        Object {
          "$attrs$": null,
          "$children$": Array [
            Object {
              "$attrs$": null,
              "$children$": null,
              "$elm$": null,
              "$flags$": 0,
              "$key$": null,
              "$name$": null,
              "$tag$": null,
              "$text$": "20",
            },
          ],
          "$elm$": null,
          "$flags$": 0,
          "$key$": null,
          "$name$": null,
          "$tag$": "span",
          "$text$": null,
        },
      ],
      "$elm$": null,
      "$flags$": 0,
      "$key$": "key1",
      "$name$": null,
      "$tag$": "li",
      "$text$": null,
    },
  ],
  "$elm$": null,
  "$flags$": 0,
  "$key$": null,
  "$name$": null,
  "$tag$": "ul",
  "$text$": null,
}
`;

exports[`qa-question-list renders no questions when none are present 1`] = `null`;
